cmd/globaldns.go

```diff
@@ -0,0 +1,931 @@
+							},
+							cli.StringFlag{
+								Name:   argAwsSecretKey,
+								EnvVar: envAwsSecretKey,
+								Usage:  "AWS secret key for route53 provider",
+							},
+							cli.StringFlag{
+								Name:   argCloudflareAPIEmail,
+								EnvVar: envCloudflareAPIEmail,
+								Usage:  "API email for Cloudflare provider",
+							},
+							cli.StringFlag{
+								Name:   argCloudflareAPIKey,
+								EnvVar: envCloudflareAPIKey,
+								Usage:  "API key for Cloudflare provider",
+							},
+						},
+					},
+					{
+						Name:      "delete",
+						Aliases:   []string{"rm"},
+						Usage:     "Delete a global DNS provider",
+						Action:    globalDNSProviderDelete,
+						ArgsUsage: "[PROVIDER_NAME/PROVIDER_ID...]",
+					},
+					{
+						Name:      "list-members",
+						Aliases:   []string{"lm"},
+						Usage:     "List members of a global DNS provider",
+						Action:    listGlobalDNSProviderMembers,
+						ArgsUsage: "[PROVIDER_NAME/PROVIDER_ID]",
+						Flags: []cli.Flag{
+							formatFlag,
+						},
+					},
+					{
+						Name:      "add-member",
+						Usage:     "Add members to a global DNS provider",
+						Action:    addGlobalDNSProviderMembers,
+						ArgsUsage: "[PROVIDER_NAME/PROVIDER_ID, USER_NAME/USER_ID...]",
+					},
+					{
+						Name:      "delete-member",
+						Usage:     "Delete members from a global DNS provider",
+						Action:    deleteGlobalDNSProviderMembers,
+						ArgsUsage: "[PROVIDER_NAME/PROVIDER_ID, USER_NAME/USER_ID...]",
+					},
+				},
+			},
+			{
+				Name:    "entries",
+				Aliases: []string{"entry"},
+				Usage:   "Operations on global DNS entries",
+				Action:  defaultAction(globalDNSLs),
+				Subcommands: []cli.Command{
+					{
+						Name:   "ls",
+						Usage:  "List global DNS entries",
+						Action: globalDNSLs,
+						Flags: []cli.Flag{
+							formatFlag,
+							quietFlag,
+						},
+					},
+					{
+						Name:   "create",
+						Usage:  "Create a global DNS entry",
+						Action: globalDNSCreate,
+						Flags: []cli.Flag{
+							cli.StringFlag{
+								Name:  argFQDN,
+								Usage: "FQDN of a global DNS entry",
+							},
+							cli.Int64Flag{
+								Name:  argTTL,
+								Usage: "DNS TTL in seconds",
+								Value: 300,
+							},
+							cli.StringFlag{
+								Name:  argProvider,
+								Usage: "Global DNS provider for an entry. Run \"rancher globaldns provider ls\" to see available ones",
+							},
+							cli.StringFlag{
+								Name:  argMultiClusterApp,
+								Usage: "Set a multi-cluster app as the target to which a global DNS entry resolves",
+							},
+							cli.StringSliceFlag{
+								Name:  argProject,
+								Usage: "Set projects as the target to which a global DNS entry resolves, can be used multiple times",
+							},
+							cli.StringSliceFlag{
+								Name:  argMember,
+								Usage: "Set members of a global DNS entry, can be used multiple times",
+							},
+						},
+					},
+					{
+						Name:      "update",
```

Comment:
Why doesn't update support projects as a flag as well? Keeping the separate command for projects is fine.

Reply:
Project target is `noupdate` fields in the API and it needs to be done via addProjects/removeProjects actions